public/scripts/s3-uploads.js

```diff
@@ -1,61 +1,91 @@
-(function () {
-  document.getElementById('camera-input').onchange = function () {
-    const files = document.getElementById('camera-input').files;
-    const file = files[0];
-    if (file == null) {
-      return new Error('No file selected.');
-    }
-    getSignedRequest(file);
-  };
-}());
-
-let imageUploadPending = false;
 
-function getSignedRequest(file) {
-  const xhr = new XMLHttpRequest();
-  xhr.onreadystatechange = () => {
-    if (xhr.readyState === 4) {
-      if (xhr.status === 200) {
-        const response = JSON.parse(xhr.responseText);
-        uploadFile(file, response.signedRequest, response.url);
-        imageUploadPending = true;
-        updateTagAndHeading(response.imageId);
+function s3handling(fileSource) {
+  (function () {
+    document.getElementById(`${fileSource}-input`).onchange = function () {
+      const files = document.getElementById(`${fileSource}-input`).files;
+      const file = files[0];
+      if (file == null) {
+        return new Error('No file selected.');
+      }
+      if (navigator.onLine) {
+        getSignedRequest(file);
+      } else if (localStorage.getItem(`toSaveFrom${fileSource}`) && localStorage.getItem(`${fileSource}TagAndHeading`)) {
+        alert("Sorry - you can only add one photo/audio memory while you're offline");
       } else {
-        return new Error('Could not get signed URL.');
+        saveFileToLocalStorage(file, fileSource);
+        updateTagAndHeading();
       }
+    };
+  }());
+
+  let uploadPending = false;
+
+  function getSignedRequest(file) {
+    const xhr = new XMLHttpRequest();
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === 4) {
+        if (xhr.status === 200) {
+          const response = JSON.parse(xhr.responseText);
+          uploadFile(file, response.signedRequest, response.url);
+          uploadPending = true;
```

Comment:
Haha - this was all the code that was merged before!! I will take a look and refactor